### PR TITLE
[server] Remove UserDeletionService EE

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

* Already checks if paymentEnabled
* Merged with the regular user deletion service

## Related Issue(s)

Fixes #

#### Preview status

gitpod:summary

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/server/src/user/user-deletion-service.ts [hunk 1]
```diff
@@ -47,6 +50,22 @@ export class UserDeletionService {
             log.debug({ userId: id }, "Is deleted but markDeleted already set. Continuing.");
         }
 
+        if (this.config.enablePayment) {
```
Comment (geropl) [inline components/server/src/user/user-deletion-service.ts:53]: This should be governed by BillingMode. But out-of-scope here, and works as well atm. :+1:

**Review by geropl — APPROVED** [review]
:+1:

File: components/server/src/user/user-deletion-service.ts (hunk 1 above)
Comment (easyCZ) [reply components/server/src/user/user-deletion-service.ts:53]: [ticketed](https://linear.app/gitpod/issue/WEB-191/control-stripe-subscription-deletion-with-billingmode)

**Pull request merged**